packages/kbn-securitysolution-io-ts-list-types/src/common/default_import_comments_array/index.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as t from 'io-ts';
+import { Either } from 'fp-ts/lib/Either';
+import { importComment, ImportCommentsArray } from '../import_comment';
+
+/**
+ * Types the ImportCommentsArray as:
+ *   - If null or undefined, then a default array of type entry will be set
+ */
+export const DefaultImportCommentsArray = new t.Type<
+  ImportCommentsArray,
+  ImportCommentsArray,
+  unknown
+>(
+  'DefaultImportComments',
+  t.array(importComment).is,
+  (input): Either<t.Errors, ImportCommentsArray> =>
+    input == null ? t.success([]) : t.array(importComment).decode(input),
```

Comment:
Instead of `.decode` here what you want is to pass in the `context` and use `validate like so:

```ts
/**
 * Types the ImportCommentsArray as:
 *   - If null or undefined, then a default array of type entry will be set
 */
export const DefaultImportCommentsArray = new t.Type<
  ImportCommentsArray,
  ImportCommentsArray,
  unknown
>(
  'DefaultImportComments',
  t.array(importComment).is,
  (input, context): Either<t.Errors, ImportCommentsArray> => // <--- Notice my context as an argument
    input == null ? t.success([]) : t.array(importComment).validate(input, context), // <-- Notice it being passed in
  t.identity
);
```

Then you run your unit tests you're going to see something much better in the output when it fails that is less terse:
```sh
node scripts/jest.js packages/kbn-securitysolution-io-ts-list-types/src/common/default_import_comments_array
```
```jest
  -   "Invalid value \"some string\" supplied to \"Array<(({| comment: NonEmptyString, created_at: string, created_by: string, id: NonEmptyString |} & Partial<{| updated_at: string, updated_by: string, meta: (object | undefined) |}>) | ({| comment: NonEmptyString |} & Partial<{| meta: (object | undefined) |}>))>\"",
  +   "Invalid value \"some string\" supplied to \"DefaultImportComments\"",
```

<img width="2269" alt="Screen Shot 2022-02-14 at 9 39 37 AM" src="https://user-images.githubusercontent.com/1151048/153907140-81216879-767e-46f5-93d8-773662fe9b4a.png">

That's how people online write their custom types. Using `validate` instead of `decode`